src/sentry/models/orgauthtoken.py

```diff
@@ -80,3 +81,23 @@ def is_org_auth_token_auth(auth: object) -> bool:
     if isinstance(auth, AuthenticatedToken):
         return auth.kind == "org_auth_token"
     return isinstance(auth, OrgAuthToken)
+
+
+def get_org_auth_token_id_from_auth(auth: object) -> int | None:
+    from sentry.services.hybrid_cloud.auth import AuthenticatedToken
+
+    if isinstance(auth, OrgAuthToken):
+        return auth.id
+    if isinstance(auth, AuthenticatedToken):
+        return auth.entity_id
+    return None
+
+
+def update_org_auth_token_last_used(auth: object, project_ids: list[int]):
+    org_auth_token_id = get_org_auth_token_id_from_auth(auth)
+    if org_auth_token_id is not None:
+        orgauthtoken_service.update_org_auth_token(
```

Comment:
This will work, but when we move to a multi-region deployment this method will become a blocking RPC call which will add upwards of 300ms to each call site. I think what you want instead for this is an outbox based delivery which is done in the background and will be eventually consistent.

We have a similar scenario for audit logs. See https://github.com/getsentry/sentry/blob/741044d03ec4239839cf6a29bce73d59eaf3fcab/src/sentry/services/hybrid_cloud/log/service.py#L39-L63

for how the silo selection logic works and https://github.com/getsentry/sentry/blob/741044d03ec4239839cf6a29bce73d59eaf3fcab/src/sentry/services/hybrid_cloud/log/impl.py#L11-L92

For how the two services are implemented. I'm happy to pair on this if you'd like.